### PR TITLE
Separate molecules into bins for selection

### PR DESCRIPTION
**Pull request by btjanaka**
DanceSelector can now separate molecules into bins based on total bond
order around the atoms, as well as fingerprint. However, it has yet to
sort by size and make the final selections.

The following changes were made:
- DanceSelector was updated to add fingerprints and sort the molecules
  into bins by the fingerprints
- danceprops was updated with a DanceFingerprint class to allow the
  fingerprinting of the molecules
- dance.py was given command line arguments to provide more parameters
  to DanceSelector
- README documentation was updated

### CODE REVIEW

File: dancelib/danceselector.py [hunk 1]
```diff
@@ -41,3 +62,50 @@ def get_data(self) -> ([oechem.OEMol], [danceprops.DanceProperties]):
...
+                self._properties).fingerprint = danceprops.DanceFingerprint(
+                    tri_n, self._wiberg_precision)
+            logging.debug(f"Added fingerprint to {mol.GetTitle()}")
+
+    def _separate_into_bins(self):
+        """
+        Separates the molecules into bins by Wiberg bond order; within each bin,
+        molecules are further separated by fingerprints.
+        """
+        for mol in self._mols:
```
Comment (jmaat) [inline dancelib/danceselector.py]: Could you add more comments into this function?

File: dancelib/danceprops.py [hunk 2]
```diff
@@ -36,8 +36,7 @@
 class DanceProperties:
```
Comment (jmaat) [inline dancelib/danceprops.py]: Could you add a comment somewhere in this script, danceprops.py (where you find most appropriate, maybe near the top where you define the data tags), an example of data that would be stored in the respective data tags?

Comment (btjanaka) [reply dancelib/danceprops.py]: I added an example on usage of get_dance_property.

File: dancelib/danceselector.py (hunk 1 above)
Comment (btjanaka) [reply dancelib/danceselector.py]: I added some comments similar to the ones we talked about on Slack.

**Pull request merged**